File: Baekjoon_Online_Judge/NO_02890/2890.cpp
#include <iostream>
using namespace std;
/*
 ---------------------------- 
 | Created by happiness96   | 
 | Year 2020                | 
 | Month 07                 | 
 | Day 28                   | 
 | 2890 카약                 | 
 ---------------------------- 
*/

int main(){
    int R, C, rank = 1, res[10] = {0,};
    char board[51][51];

    cin >> R >> C;

    for(int i = 0; i < R; i ++){
        for(int j = 0; j < C; j ++) cin >> board[i][j];
    }

    for(int col = C - 2; col > 0; col --){
        int flag = 0;

        for(int row = 0; row < R; row ++){
            char val = board[row][col];
            
            if('1' <= val <= '9'){
                if(res[(int)val - '0'] == 0){
                    res[(int)val - '0'] = rank;
                    flag = 1;
                }
            }
        } 
        rank += flag;
    }

    for(int i = 1; i < 10; i ++) cout << res[i] << "\n";

    return 0;
}

File: Baekjoon_Online_Judge/NO_02890/2890.py
# -*- encoding: utf-8 -*-
import sys
r_input = sys.stdin.readline

# ---------------------------- #
# | Created by happiness96   | #
# | Year 2020                | #
# | Month 07                 | #
# | Day 28                   | #
# | 2890 카약                 | #
# ---------------------------- #


if __name__ == "__main__":
    R, C = map(int, r_input().split())

    res = [0] * 10
    rank = 1

    board = [list(r_input().rstrip()) for _ in range(R)]

    for col in range(C - 2, 0, -1):
        flag = 0

        for row in range(R):
            val = board[row][col]

            if val in '123456789':
                if not res[int(val)]:
                    flag = 1

                    res[int(val)] = rank
        
        rank += flag
    
    print(*res[1:], sep='\n')
